runtime/near-vm-runner/src/tests/fuzzers.rs

```diff
@@ -167,3 +170,31 @@ fn wasmer2_and_wasmtime_agree() {
         assert_eq!(wasmer2, wasmtime);
     });
 }
+
+#[test]
+fn wasmer2_is_reproducible() {
```

Comment:
I see we also attach a crash file here `crash-7efd70f50949e0fdff8f4c8d6cc66709faf6d3cf`. I am a tiny bit skeptical of the utility of them -- it seems it's relatively easy for the crash file to become obsolete with unrelated changes. 

I *think* what is in the crash file is `data: &[u8]`, the seed we pass to arbitrary. But that means that, eg, upgrading `arbitrary` can chance the `.wasm` that is generated, which might break the test. 

It seems to me that, for generative fuzzing `[u8] -> T` we should save `T`, rather than `[u8]`, as a crash. So, in this case, an actual `.wasm`. 

Action items here: 

* this probably wants some design discussion on the bolero repository
* I think I am fine with including the current crash file here, but I am curious whether the default 1000 iterations would be enough to get it? If it's easy to discover, I suggest not adding a file maybe.

Reply:
While what you say makes sense, what would be a better option in your opinion? I don’t think we can do this straight from bolero, because we’d have to have _some_ serialization method for storing the crash report, and Arbitrary is as good a serialization method as any other afaict.

The alternative I can think of that would make sense would be to just add a unit test instead of checking in a crash report. I didn’t go that way first, but could switch to this method if you think it’d be better?

Reply:
Oh and, it took my computer approx. 20min/1hr to find the crash from scratch, so unfortunately I don’t think just not adding an explicit test is a solution. (Also, I let the computer search for a few more hours after that and didn’t find any other determinism issue, so hopefully the rest of our code is ok-ish)

Reply:
>Oh and, it took my computer approx. 20min/1hr to find the crash from scratch

Yeah, that's a good argument for keeping this as is. 


> I don’t think we can do this straight from bolero, because we’d have to have some serialization method for storing the crash report, and Arbitrary is as good a serialization method as any other afaict.

That's the crux of my thinking here -- I do think that `Arbitrary` is less good serialization method, because it is meaningfully less stable. Specifically: 

* if we store raw bytes, and do `cargo upgrade -p wasm-smith`, the test could break
* if we store wasm generated by arbitrary *by some particualar version*  of `wasm-smith`, than the test would be reproducible even after upgrades. 

A slightly different way to get at this: it might be the case that some `.wasm`s are impossible to generate via `wasm-smith`, while raw `.wasm` naturally can represent any `.wasm`. The generation routine is not necessary surjective, while proper serialization is. 

OTOH, I also don't see small-incremental solutions here. A "large" solution would be something like the following

```rust
traint TestInput
  : arbitrary::Arbitrary // to generate from raw bytes
  + Display // to show on failures
{
  // to save/load examples independent from arbitrary,
  // or for manual table-driven tests. 
  fn dump(&self) -> Vec<u8> 
  fn load(Vec<u8>) -> Result<Self>
}
```

Reply:
Hmm so I’m not entirely sure how that’d work out, because even Arbitrary is but one of the ways to generate inputs with bolero, there’s also bolero’s own TypeGenerator and ValueGenerator, and I think there are even other generators they’re considering adding.

That said what you’re saying does make sense to me: it’d be good to be able to provide a "value" instead of a fuzzer seed, if only so that it becomes possible to use a known crash input for the fuzzer. I had opened https://github.com/camshaft/bolero/issues/91 to track this, though I’m not sure how doable-in-practice that’d be. Does it make sense to you?